// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package math;

// The greeting service definition.
service MathOperations {
  // Performs multiplication for two numbers
  rpc Multiplication (Multiplicands) returns (Result) {}
  // Performs nth Triangular Addition (factorial for addition) through streaming
  rpc nthTriangularNumber (stream Addends) returns (Result) {}
  // Takes streamed Addends, returns streamed addends + 5
  rpc AdditionPlusFive (stream Addends) returns (stream Result) {}
}

// The request message containing the user's name.
message Multiplicands {
  int32 multiplicand1 = 1;
  int32 multiplicand2 = 2;
}

message Addends {
  int32 addend = 1;
}

message Result {
  int64 mathResult = 1;
}
